torchx/schedulers/kueue_job_scheduler.py

```diff
@@ -0,0 +1,809 @@
+
+    def _api_client(self) -> "ApiClient":
+        from kubernetes import client, config
+
+        c = self._client
+        if c is None:
+            configuration = client.Configuration()
+            try:
+                config.load_kube_config(client_configuration=configuration)
+            except config.ConfigException as e:
+                warnings.warn(f"failed to load kube config: {e}")
+
+            c = self._client = client.ApiClient(configuration)
+
+        return c
+
+    def _custom_objects_api(self) -> "CustomObjectsApi":
+        from kubernetes import client
+
+        return client.CustomObjectsApi(self._api_client())
+
+    def _batchv1_api(self) -> "BatchV1Api":
+        from kubernetes import client
+
+        return client.BatchV1Api(self._api_client())
+
+    def _corev1_api(self) -> "CoreV1Api":
+        from kubernetes import client
+
+        return client.CoreV1Api(self._api_client())
+
+    def _get_job_name_from_exception(self, e: "ApiException") -> Optional[str]:
+        try:
+            return json.loads(e.body)["details"]["name"]
+        except Exception as e:
+            logger.exception("Unable to retrieve job name, got exception", e)
+            return None
+
+    def _get_active_context(self) -> Dict[str, Any]:
+        from kubernetes import config
+
+        contexts, active_context = config.list_kube_config_contexts()
+        return active_context
+
+    def schedule(self, dryrun_info: AppDryRunInfo[KueueJob]) -> str:
+        from kubernetes.client.rest import ApiException
+
+        cfg = dryrun_info._cfg
+        assert cfg is not None, f"{dryrun_info} missing cfg"
+        namespace = cfg.get("namespace") or "default"
+
+        images_to_push = dryrun_info.request.images_to_push
+        self.push_images(images_to_push)
+
+        resource = dryrun_info.request.resource
+        try:
+            resp = self._batchv1_api().create_namespaced_job(
+                namespace=namespace,
+                body=resource,
+            )
+
+        except ApiException as e:
+            if e.status == 409 and e.reason == "Conflict":
+                job_name = self._get_job_name_from_exception(e)
+                raise ValueError(
+                    f"Job `{job_name}` already exists. This seems like a transient exception, try resubmitting job"
+                ) from e
+            else:
+                raise
+
+        return f'{namespace}:{resp.metadata.name}'
+
+    def _submit_dryrun(
+        self, app: AppDef, cfg: KueueJobOpts
+    ) -> AppDryRunInfo[KueueJob]:
+        # map any local images to the remote image
+        images_to_push = self.dryrun_push_images(app, cast(Mapping[str, CfgVal], cfg))
+
+        service_account = cfg.get("service_account")
+        assert service_account is None or isinstance(
+            service_account, str
+        ), "service_account must be a str"
+
+        local_kueue = cfg.get("local_kueue")
+        assert local_kueue is None or isinstance(
```

Comment:
You can remove `local_kueue is None` from this conditional that way you don't need to check for it in `app_to_resource`